File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Worldbossaward.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Worldbossaward implements ITableOperate {
 public final static String TAB_FILE_DATA = "/worldbossaward.txt";
 public static final int ID_TYPE = 0 ;
public static final int ID_PAR = 1 ;
public static final int ID_AWARD_GOLD = 2 ;
public static final int ID_AWARD_YUANBAO = 3 ;
public static final int ID_AWARD_LEADER = 4 ;
public static final int MAX_RECORD = 5 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_AwardGold;
 public int GetAwardGold() { return m_AwardGold;}

private int m_AwardLeader;
 public int GetAwardLeader() { return m_AwardLeader;}

private int m_AwardYuanbao;
 public int GetAwardYuanbao() { return m_AwardYuanbao;}

private int m_Par;
 public int GetPar() { return m_Par;}

private int m_Type;
 public int GetType() { return m_Type;}

public boolean LoadTable(HashMap<Integer, Table_Worldbossaward> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Worldbossaward> _hash = (HashMap<Integer, Table_Worldbossaward> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Worldbossaward _values = new Table_Worldbossaward();
 _values.m_AwardGold =  Integer.parseInt((String)valuesList.get(ID_AWARD_GOLD));
_values.m_AwardLeader =  Integer.parseInt((String)valuesList.get(ID_AWARD_LEADER));
_values.m_AwardYuanbao =  Integer.parseInt((String)valuesList.get(ID_AWARD_YUANBAO));
_values.m_Par =  Integer.parseInt((String)valuesList.get(ID_PAR));
_values.m_Type =  Integer.parseInt((String)valuesList.get(ID_TYPE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Worldbossaward.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Worldbossaward : ITableOperate
{ private const string TAB_FILE_DATA = "worldbossaward.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_TYPE,
ID_PAR,
ID_AWARD_GOLD,
ID_AWARD_YUANBAO,
ID_AWARD_LEADER,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_AwardGold;
 public int AwardGold { get{ return m_AwardGold;}}

private int m_AwardLeader;
 public int AwardLeader { get{ return m_AwardLeader;}}

private int m_AwardYuanbao;
 public int AwardYuanbao { get{ return m_AwardYuanbao;}}

private int m_Par;
 public int Par { get{ return m_Par;}}

private int m_Type;
 public int Type { get{ return m_Type;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Worldbossaward _values = new Tab_Worldbossaward();
 _values.m_AwardGold =  Convert.ToInt32(valuesList[(int)_ID.ID_AWARD_GOLD] as string);
_values.m_AwardLeader =  Convert.ToInt32(valuesList[(int)_ID.ID_AWARD_LEADER] as string);
_values.m_AwardYuanbao =  Convert.ToInt32(valuesList[(int)_ID.ID_AWARD_YUANBAO] as string);
_values.m_Par =  Convert.ToInt32(valuesList[(int)_ID.ID_PAR] as string);
_values.m_Type =  Convert.ToInt32(valuesList[(int)_ID.ID_TYPE] as string);

 _hash.Add(nKey,_values); }


}
}

